apps/examples/log_dump/log_dump_main.c

```diff
@@ -0,0 +1,49 @@
+/****************************************************************************
+ *
+ * Copyright 2022 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+/****************************************************************************
+ * log_dump_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int log_dump_main(int argc, char *argv[])
+#endif
+{
+
+	int fd = open("/proc/logsave", O_RDONLY);
+	int ret = 1;
+	char buf[CONFIG_LOG_DUMP_CHUNK_SIZE];
+	while (ret > 0) {
+     		ret = read(fd, buf, sizeof(buf));
+		for (int i = 0; i < ret; i++) {
+			lldbg_noarg("%c", buf[i]);
```

Comment:
Could you check below?
1. how about printing with %s?
2. how about using printf instead lldbg?

If not, could you let me know the reason?

Reply:
1. buf is not always completely filled. While reading the last block, the return value can be less than the requested read size. As a result, if we use %s garbage value will be printed at the end. I can add '\0' at the end for such cases at the end and printf using '%s'. I will change it to "%s"
2. Assume the case where the last block is just read, ret > 0, followed by printf lines. these again gets added to the log buffer and in the next iteration, we get a non 0 value for ret as new data is added. So it will go into infinite loop if we try to read until ret == 0. So I used lldbg as they will not be added to the saved logs.

Reply:
1. Ok. You're right. Please ignore my comment for this. (Please don't add \0)

Reply:
2. Sorry I don't get your point. Even next data is remain and while loop reads it again and again, why we can't use printf? As lldbg_noarg is not common way to print logs, I leave this comment.

Reply:
every time we use printf to print the data in the while loop it gets added to the saved logs. So the size of the saved logs will never be zero resulting in an infinite loop. If we use lldbg, these logs are not added to the saved logs. Hence the saved logs will finally give the read size to be zero once the entire logs are read. To avoid the infinite loop, I used lldbg. If we use printf, it will be similar to trying to completely read a file in which we keep on writing data.